NikoBracket/CreateBracketViewController.m

```diff
@@ -175,24 +411,136 @@ - (UIView *)tableView:(UITableView *)tableView viewForHeaderInSection:(NSInteger
         header.textLabel.text = @"Midwest";
     }
     
+    if (self.roundControl.selectedSegmentIndex > 3) {
+        header.textLabel.text = @"";
+    }
+    
     header.textLabel.font = [header.textLabel.font fontWithSize:30];
     
     return header;
 }
 
+- (NSMutableArray *) initializeUserBracket:(NSMutableArray *) matchups {
+    NSMutableArray *round1 = matchups[0];
+    NSMutableArray *round2 = [[NSMutableArray alloc] init];
+    NSMutableArray *round3 = [[NSMutableArray alloc] init];
+    NSMutableArray *round4 = [[NSMutableArray alloc] init];
     
+    for (int i = 0; i < round1.count - 1; i+=2) {
+        NSArray *game = round1[i];
+        NSArray *game2 = round1[i+1];
 
+        NSString *winner1 = game[[game[2] intValue]];
+        NSString *winner2 = game2[[game2[2] intValue]];
+        if (matchups.count == 1) {
+            NSMutableArray *matchup = [[NSMutableArray alloc] initWithObjects:winner1, winner2, @"0", nil];
+            [round2 addObject:matchup];
+        }
+        else{
+            NSString *choice = matchups[1][i/2][2];
+            NSMutableArray *matchup = [[NSMutableArray alloc] initWithObjects:winner1, winner2, choice, nil];
+            [round2 addObject:matchup];
+        }
+       
+    }
+    for (int i = 0; i < round2.count - 1; i+=2) {
+        NSArray *game = round2[i];
+        NSArray *game2 = round2[i+1];
 
+        NSString *winner1 = game[[game[2] intValue]];
+        NSString *winner2 = game2[[game2[2] intValue]];
+        if (matchups.count == 1) {
+            NSMutableArray *matchup = [[NSMutableArray alloc] initWithObjects:winner1, winner2, @"0", nil];
+            [round3 addObject:matchup];
+        }
+        else{
+            NSString *choice = matchups[2][i/2][2];
+            NSMutableArray *matchup = [[NSMutableArray alloc] initWithObjects:winner1, winner2, choice, nil];
+            [round3 addObject:matchup];
+        }
+    }
+    for (int i = 0; i < round3.count - 1; i+=2) {
+        NSArray *game = round3[i];
+        NSArray *game2 = round3[i+1];
 
-/*
-#pragma mark - Navigation
+        NSString *winner1 = game[[game[2] intValue]];
+        NSString *winner2 = game2[[game2[2] intValue]];
+        if (matchups.count == 1) {
+            NSMutableArray *matchup = [[NSMutableArray alloc] initWithObjects:winner1, winner2, @"0", nil];
+            [round4 addObject:matchup];
+        }
+        else{
+            NSString *choice = matchups[3][i/2][2];
+            NSMutableArray *matchup = [[NSMutableArray alloc] initWithObjects:winner1, winner2, choice, nil];
+            [round4 addObject:matchup];
+        }
+    }
+    
+    matchups[1] = round2;
+    matchups[2] = round3;
+    matchups[3] = round4;
```

Comment:
I think this may just be accidentally working. We should check the size of the `matchups` array, and if its not big enough then handle this case